cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -165,7 +166,15 @@ func getAbapCommunicationArrangementInfo(config abapEnvironmentPullGitRepoOption
 
 	if config.Host != "" {
 		// Host, User and Password are directly provided
-		connectionDetails.URL = "https://" + config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+		matchedkey, error := regexp.MatchString("[hH][tT][tT][pP][sS]://*", config.Host)
```

Comment:
I never saw `https:` provided in upper case. May I ask for the reason? The scheme [index](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml) only uses lower case characters.

What about plain `http`? There are reasons for not using it and `https` is state of the art. But in case any customer would like to use plain `http`, why should we prevent him from doing so? (by the way: yes, I saw there was an explicit check for `https` also before.

I would think twice before merging this.

Reply:
Regarding the regex:
  - should't it be`.*` rather than `*` in `[hH][tT][tT][pP][sS]://*`
  - is there no need for escaping the slashes ? (`https:\/\/` rather than `https://`)
  - I guess the regex would also match e.g. `xhttps://example.org` (mind the 'x').

Reply:
Regarding `https:` provided in upper case: True, usually users wouldn't type `https:` in upper cases. However, if the user provides it in upper cases this will still get handled. This was the reason behind handling upper cases.

Regarding plain `http`: Since Steampunk systems are only reachable via `https` protocol this doesn't get handled here. Do you think it would be beneficial to catch when user wants to use `http` protocol and change it to `https`?

Reply:
> it in upper cases this will still get handled

seems to be not aligned with the list of assigned schemes (there only lower case is used), but ok ...

> Do you think it would be beneficial to catch [...]

no. Such kind of "we know better what the user wants or should do" makes more trouble that benefits.

Reply:
I changed the Regex to `^[hH][tT][tT][pP][sS]:\/\/.*`. 
Please let me know what you think.